### PR TITLE
Automated cherry pick of #16287 #16546 #16565

### PR DESCRIPTION
**Pull request by j3ffml**
Cherry pick of #16287 #16546 #16565 on release-1.1.

### CODE REVIEW

**Comment by jlowdermilk** [comment]
#16551 take 2

**Comment by jlowdermilk** [comment]
@k8s-bot test this

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this

File: hack/jenkins/test-dockerized.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export KUBE_KEEP_VERBOSE_TEST_OUTPUT=y
+export KUBE_TIMEOUT='-timeout 300s'
+export KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4
+export LOG_LEVEL=4
+export KUBE_TEST_API_VERSIONS=v1,extensions/v1beta1
+export KUBE_TEST_ETCD_PREFIXES=registry
+
+./hack/build-go.sh
+godep go install ./...
+./hack/travis/install-etcd.sh
```
Comment (jlowdermilk) [inline hack/jenkins/test-dockerized.sh]: Commenting here to note the cherrypick wasn't fully automated. This file has some differences between the one at HEAD to deal with moved script locations and branch-specific test env.

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this

**Comment by ixdy** [comment]
LGTM

@k8s-bot unit test this please

**Comment by jlowdermilk** [comment]
I think the cni test failure is #14931

**Comment by a-robinson** [comment]
Manual merge while merge-bot is down due to a single flake from a known issue.

**Pull request merged**